### ISSUE TITLE
Persistence Module (universal-localstorage) have problems with the nodejs

### ISSUE THREAD

**Issue by vitormsilva** [reTHINK-project/dev-runtime-core#95]
Regarding all the problems the runtime-nodejs developer team was facing, we did an evaluation and found some problems related with de [universal-localstorage module](https://github.com/eddiemoore/universal-localstorage), is not maintained and updated. The function [fs.existsSync](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback) was deprecated from most recent node versions, since 1.0.0;

So, I suggest @dvilchez you found other solution for this module.

By the way, I suggest to, pass in the runtimeFactory the persistanceModule, to be used in each environment, since the browser have it natively, and nodejs have this module [node-localstorage](https://github.com/lmaccherone/node-localstorage), i don't know, is only a suggestion..

This affects to the dev-service-framework, should be fixed as soon as possible;

**Comment by jboulmal** [comment]
We have gone through this problem in this [issue Local Storage in Node.js]( https://github.com/reTHINK-project/dev-service-framework/issues/46).We  have fixed it by installing the node-module [universal-storage, which is supposed to provide localStorage both in browser and nodejs](https://github.com/eddiemoore/universal-localstorage) that was proposed by @Endebert.

**Comment by dvilchez** [comment]
universal-localstorage has a very old version of nodejs-localstorage, this version of nodejs-local-storage is using some deprecated functions on fs module, but it works. I agree with you we have to update this dependency but it is not the problem.

The problem here is to use a browserify distribution file from nodejs. Browserify was thought to use nodejs modules from browser and not in the other way. You could use a UMD distribution file but runtime-core and service-frameworks are not defined as universal modules.

**Comment by NicoApizee** [comment]
Same conclusion here https://github.com/reTHINK-project/dev-runtime-core/issues/94.

**Comment by vitormsilva** [comment]
If you look at browserify documentation, https://github.com/substack/node-browserify#usage, when you pass the standalone option, the modules are created like a UMD module;

If you look at [Gulpfile service-framework](https://github.com/reTHINK-project/dev-service-framework/blob/develop/gulpfile.js#L292) on dev-service-framework all the modules are exported with UMD format;

If you look at [Gulpfile runtime-core](https://github.com/reTHINK-project/dev-runtime-core/blob/master/gulpfile.js#L220)

Why you telling the problem is the browserify? Have I some options wrong? can you help to optimize this process?

**Comment by NicoApizee** [comment]
It also provide option for node :
  --no-browser-field, --no-bf
    Turn off package.json browser field resolution. This is also handy if you
    need to run a bundle in node.

  --node
    Alias for --bare and --no-browser-field.

But I'm not sure that the same "build" can be run on both browser & node.

**Comment by dvilchez** [comment]
@vitormsilva I was reviewing browserify UMD implementation, and I am going to make a guess, even you set "UMD"option browserify is making a bundle file, including fs lib but for some reason it doesn't include  deprecated methods on this module.

I never have used a bundle file as a modules on NodeJS, I not even know if it is a good or bad practice. But if you want to use it in this way I think you have to use --node parameter on browserify build, and maybe even in this way doesn't use deprecated methods is mandatory.

**Comment by vitormsilva** [comment]
Hi,

I think the problem isn't the the UMD distribution file, because, when you pass in the standalone option you don't care about the nodejs or browser, is UMD, should work; (API)[https://github.com/substack/node-browserify#browserifyfiles--opts]

I generate an runtime-core file [Runtime.txt](https://github.com/reTHINK-project/dev-runtime-core/files/339166/Runtime.txt) and i did search for fs.existsSync, and the result is only 3 places here: [fs exists.txt](https://github.com/reTHINK-project/dev-runtime-core/files/339165/fs.exists.txt)

You can do the same test, I uploaded the Runtime.txt for you can test;

**Comment by dvilchez** [comment]
ok, as you want. Anyway I did the changes this morning 1b54eb5f67044728d24e3f45abc4aa4a5a4b3ba9. I only let this issue open because I think is important to discuss the right way to distribute rethink and its components because it could be a problem for nodejs runtime.

**Comment by pchainho** [comment]
can we close this?

**Comment by jboulmal** [comment]
I agree with @dvilchez,  we are strongly depending on enhancement of the develop branche of runtime-core.

**Comment by pchainho** [comment]
Pls open issues for these enhancements
